content/technotes/2018-09-03-commonmark.md

```diff
@@ -0,0 +1,327 @@
+---
+slug: "commonmark"
+title: "In praise of Commonmark: wrangle (R)Markdown files without regex"
+date: 2018-09-03
+authors:
+  - name: Maëlle Salmon
+    url: https://masalmon.eu/
+categories:
+  - technotes
+topicid: 1307
+tags:
+- Markdown
+- RMarkdown
+- xml2
+- commonmark
+output: 
+  md_document:
+    variant: markdown_github
+    preserve_yaml: true
+---
+
+You might have read [my blog post analyzing the social weather of
+rOpenSci
+onboarding](https://ropensci.org/blog/2018/05/10/onboarding-social-weather/),
+based on a text analysis of GitHub issues. I extracted text out of
+Markdown-formatted threads with regular expressions. I basically
+hammered away at the issues using tools I was familiar with until it
+worked! Now I know there’s a much better and cleaner way, that I’ll
+present in this note. Read on if you want to extract insights about
+text, code, links, etc. from R Markdown reports, Hugo website sources,
+GitHub issues… without writing messy and smelly code!
+
+### Introduction to Markdown rendering and parsing
+
+This note will appear to you, dear reader, as an html page, either here
+on ropensci.org or on R-Bloggers, but I’m writing it as an R Markdown
+document, using Markdown syntax. I’ll knit it to Markdown and then
+Hugo’s Markdown processor,
+[Blackfriday](https://github.com/russross/blackfriday), will transform
+it to html. Elements such as “\# blabla” thus get transformed to
```

Comment:
Maybe `# blabla` can be syntax as well (backticks instead of dquotes)

Reply:
Ah good point!